subworkflows/local/busco_qc.nf

```diff
@@ -9,23 +9,57 @@ include { BUSCO_SUMMARY                   } from '../../modules/local/busco_summ
 
 workflow BUSCO_QC {
     take:
-    busco_db_file           // channel: path
-    busco_download_folder   // channel: path
-    bins                    // channel: [ val(meta), path(bin) ]
+    busco_db   // channel: path
+    bins       // channel: [ val(meta), path(bin) ]
 
     main:
-    if (params.busco_reference){
-        BUSCO_DB_PREPARATION ( busco_db_file )
-        ch_busco_db = BUSCO_DB_PREPARATION.out.db
+    if ( !busco_db.isEmpty() ) {
+        if ( busco_db.extension == "gz" ) {
+            // Expects to be tar.gz!
+            BUSCO_DB_PREPARATION ( busco_db )
+
+            ch_db_for_busco = BUSCO_DB_PREPARATION.out.db
+                                .map{
+                                    meta, db ->
+                                        def meta_new = [:]
+                                        meta_new['id'] = meta
+                                        meta_new['lineage'] = 'Y'
+                                        [ meta_new, db ]
+                                }
+        } else if ( busco_db.isDirectory() ) {
+            // Set meta to match expected channel cardinality for BUSCO
+            ch_db_for_busco = busco_db
+                                .map{
+                                    db ->
+                                        def meta = [:]
+                                        meta['id'] = db.toString().split('/').last()
```

Comment:
I think ti would be safer to use one of the file attribute functions: https://www.nextflow.io/docs/latest/script.html#check-file-attributes